content/docs/buildpack-author-guide/caching-strategies.md

```diff
@@ -0,0 +1,65 @@
++++
+title="Caching Strategies"
+weight=6
+summary="Learn strategies for caching layers."
++++
+
+# Caching
+
+There are three types of layers that can be contributed to an image
+
+* `build` layers -- the directory will be accessible by subsequent buildpacks,
+* `cache` layers -- the directory will be included in the cache,
+* `launch` layers -- the directory will be included in the run image as a single layer,
+
+A fourth type of layer
+
+* `ignored` layers
+
+are available to buildpack authors for use as temporary layers.
+
+In this section we look at caching each layer type.
+
+## Layer Metadata
+
+Buildpacks ensure byte-for-byte reproducibility of layers.  File creation time is normalized to January 1, 1980 to ensure reproducibility.  Byte-for-byte reproducibility means we can re-use previous layers.  However, we want to invalidate previously cached layers if
+
+* the buildpacks API changes,
+* the type of the layer changes.
+
+A layer built using a buildpack at API version `0.7` should be considered invalid if the API version for that buildpack has been updated to `0.8`.  Similarly, if a layer is changed from being a cache-only layer to being a cache and launch layer, then the cache should be considered invalid.
+
+In addition to general cache invalidation conditions our pip buildpack should invalidate a previous layer if an important property changes, such as:
```

Comment:
Why are you mentioning a pip buildpack specifically?

Reply:
Thanks.  This was a mistake.